### ISSUE TITLE
"tenant_subfolder_tutorial" project example not working

### ISSUE THREAD

**Issue by infosrabah** [django-tenants/django-tenants#546]
Hi,

I worked with django-tenants for subdomain and it works great, heard that dango-tenants offers now URL subflder and I gave it a try, directly on the 'tenant_subfolder_tutorial' project packed by django-tenants. but ...

While going to public url routes and tenant url routes it's not working it display and error " "'WSGIRequest' object has no attribute 'tenant'".

#####################
The process I've made :
 - I cloned the repo
 - python3 manage.py makemigrations customers tenant_only
 - python3 manage.py migrate_schemas --shared

- created the public tenant :
```python
# Public tenant
from customers.models import Client, Domain
tenant = Client(schema_name='public', name='Schemas Inc.')
tenant.save()
domain = Domain()
domain.domain = 'my-domain.local''
domain.tenant = tenant
domain.is_primary = True
domain.save()
```

- created the tenant :
```python
# Tenant
from customers.models import Client, Domain
tenant = Client(schema_name='tenant1', name='Fonzy Tenant')
tenant.save()
domain = Domain()
domain.domain = 'tenant.my-domain.local' 
domain.tenant = tenant
domain.is_primary = True
domain.save()
```
#####

After accessing tenant at my local server with : 127.0.0.1:8000/clients/tenant1

it shows me the error :  "'WSGIRequest' object has no attribute 'tenant'".

NB : This issue is related to "tenant_subfolder_tutorial" project

Any ideas ?

Thank you

**Comment by correadevuk** [comment]
Make sure you have set TENANT_SUBFOLDER_PREFIX, for example, TENANT_SUBFOLDER_PREFIX = "customers"

When creating your 'public' schema, run "python manage.py create_tenant", and on domain, just press enter. Now you will be able to access as 127.0.0.1:8000, for example 

After creating your first tenant, you will be able to access it as  127.0.0.1:8000/clientes/{domain}
domain can be any value that you set when creating your schema via the create_tenant command. Check the table customer_client

**Comment by infosrabah** [comment]
Hi Correadevuk,thank you for the reply.

Did all that without a luck, it still display the error as above.

do you have a repo made for that that I can use or the exact command list to do after setting the project to overcome this issue ?

to be clear I USED THE SAME EXAMPLE REPO AS IT IS, how come isn"t working,

thank you,

**Comment by correadevuk** [comment]
@infosrabah I'm using the project example that you find in the repo. You've mentioned that you are using this code to create a public user : 

` Public tenant
from customers.models import Client, Domain
tenant = Client(schema_name='public', name='Schemas Inc.')
tenant.save()
domain = Domain()
domain.domain = 'my-domain.local''
...`

What I'm trying to tell you is to do this instead : 
1 - after cloning the project, run `python manage.py migrate_schemas --shared`, you will see all the default tables created on public schema
2 - Now run : python manage.py create_tenant  

`(venv)  vitorcorrea@Vitors-Mac-mini  ~/Documents/tenant_subfolder_tutorial  python manage.py create_tenant           
model created
schema name: public
name: public
description: public schema
domain: 
is primary (leave blank to use 'True'): `

notice that I left "domain" blank, just press enter

now just access http://127.0.0.1:8000

![Screenshot 2021-03-21 at 10 08 09](https://user-images.githubusercontent.com/65387466/111900985-5fc77280-8a2d-11eb-91b9-e4d0f03d668a.png)

**Comment by infosrabah** [comment]
Hi correadevuk ,

Thank you so much for the reply..

Here's what I did :
- Deleted the db
- Delete the whole folder of django-tenans
- created the folder "django-tenans" and the db "tenant_subfolder_tutorial"
- cloned the repo
- python3 manage.py migrate_schemas --shared (as you said)
- python3 manage.py create_tenant (supplied the exact same entries as you)
...... until now everthing went well : no error, so far so good
- Python3 manage.py runserver
- http://127.0.0.1:8000 ----> I got an error as before : "'WSGIRequest' object has no attribute 'tenant'".
![django-tenants-error](https://user-images.githubusercontent.com/34539758/111937646-16743300-8ac0-11eb-828c-f31e39e9e278.png)

and here my pip3 freeze output :
asgiref==3.3.1
Django==3.1.3
django-tenants==3.2.1
gunicorn==20.0.4
psycopg2==2.8.6
pytz==2021.1
sqlparse==0.4.1

Hope all of this helps to spot what's wrong.

just to let you know that all my django projects work well, except django-tenants (I was suspecting gunicorn, but uninstall it and try...without luck the same error message)

I'm using Debian 10/XFCE 4

thank's so much correadevuk to take time to help.
